flow/Platform.h

```diff
@@ -421,6 +426,13 @@ inline static uint64_t __rdtsc() {
 }
 #endif
 #endif
+#ifdef __aarch64__
+inline static uint64_t __rdtsc() {
```

Comment:
defining `__rdtsc` can be problematic - this could be defined in newer versions of `gcc`. Names starting with double underscore `__` are reserved for the compiler. So this could break with a compiler upgrade.

Instead I would suggest the following:

1. Define a new function in `Platform.h` called `timestampCounter` (or something else if you don't like the name) like this:
```
inline static uint64_t timestampCounter() {
#ifdef __aarch64__
    uint64_t timer;
    asm volatile("mrs %0, cntvct_el0" : "=r"(timer));
    return timer;
#else
    return __rdtsc();
#endif
}
```
2. Replace all usages of `__rdtsc` with this function (there are 8 - so this should be feasible).

Reply:
Yes, I was wondering the very same thing.  I followed the OSX port.
Ideally, we should get rid of any builtins and intrinsics from everywhere and consolidate them into Platform.h or somewhere, then every platform should call the common function.
e.g. CRC functions, prefetch functions, other SIMD functions...
But it will be a rather larger project than just "ARM port".  We should do it in a separate PR, IMHO.

Reply:
I don't think this needs to be done in one go. This can very easily be done in a piece-meal way. As I wrote above, we are talking about 8 calls.

Also the current implementation is not correct - we check for `__APPLE__` on line 415 (the implementation there is also problematic, but at least it also checks `__has_builtin(__rdtsc)`). So if you would compile on `__APPLE__` and `aarch64`, `__rdtsc` will now be defined twice.

Fixing this now is imho much easier than to track those things down when we add support for additional platforms...

Reply:
Yeah, I just noticed the builtin check for APPLE.  Anyways, yeah, I agree, this is such a small function, let me fix this in this PR.